lib/views/topLevelView.js

```diff
@@ -0,0 +1,351 @@
+        }
+	meta = summarizeSBOL(typeShortName,topLevel,req,sbol,remote,graphUri)
+	
+    }).then(function lookupCollections() {
+
+        return Promise.all([
+	    getContainingCollections(uri, graphUri, req.url).then((_collections) => {
+
+		collections = _collections
+
+		collections.forEach((collection) => {
+
+		    collection.url = uriToUrl(collection.uri)
+
+                    const collectionIcons = config.get('collectionIcons')
+                    
+                    if(collectionIcons[collection.uri])
+			collectionIcon = collectionIcons[collection.uri]
+		})
+            }),
+
+	    sparql.queryJson(getCitationsQuery, graphUri).then((results) => {
+
+                citations = results
+
+            }).then(() => {
+
+                return retrieveCitations(citations).then((resolvedCitations) => {
+
+                    submissionCitations = resolvedCitations;
+
+                })
+
+            }),
+
+            sparql.queryJson(getBuildsQuery, graphUri).then((results) => {
+
+		builds = results
+		builds.forEach((build) => {
+		    build.url = uriToUrl(build.impl)
+		})
+            })
+
+        ])
+
+    }).then(function lookupEncodedProteins() {
+
+        var query =
+            'PREFIX sybio: <http://w3id.org/sybio/ont#>\n' +
+        'SELECT ?subject WHERE {' +
+            '   ?subject sybio:encodedBy <' + uri + '>' +
+            '}'
+
+        return sparql.queryJson(query, graphUri).then((results) => {
+
+            encodedProteins = results.map((result) => {
+                return result.subject
+            })
+
+        })
+
+    }).then(function getOtherComponentMetaData() {
+
+        if(meta.protein && meta.protein.encodedBy)
+            otherComponents = otherComponents.concat(meta.protein.encodedBy)
+                    /* todo and subcomponents */
+
+        otherComponents = otherComponents.concat(encodedProteins)
+
+        return Promise.all(otherComponents.map((otherComponent) => {
+
+            return getComponentDefinitionMetadata(otherComponent, graphUri).then((res) => {
+
+                mappings[otherComponent] = res.metaData.name
+
+            })
+
+        }))
+
+    }).then(function summarizeEncodedProteins() {
+
+        meta.encodedProteins = encodedProteins.map((uri) => {
+
+	    return {
+                uri: uri,
+                name: mappings[uri],
+                url: uri
+	    }
+
+        })
+
+        if(meta.protein) {
+
+	    if(meta.protein.encodedBy) {
+
+                meta.protein.encodedBy = meta.protein.encodedBy.map((uri) => {
+
+		    var prefixified = prefixify(uri, prefixes)
+
+		    return {
+                        uri: uri,
+                        name: mappings[uri],
+                        url: '/entry/' + prefixified.prefix + '/' + prefixified.uri
+		    }
+
+                })
+
+	    }
+        }
+
+    }).then(function fetchFromIgem() {
+
+	if(topLevel.wasDerivedFrom.toString().indexOf('http://parts.igem.org/') === 0) {
```

Comment:
Is this hitting the iGem registry on _every_ iGem part pageload? That's a lot, and will be slow. Could we maybe offload this onto a nightly job that updates from iGem?

Reply:
Or this should be a plugin.

Reply:
Would be good to start by moving this out of the file into its own that is only hit on CD